<launch>
    <arg name="backend" default=""/>
    <group if="$(eval arg('backend') != '')">
        <node pkg="rasberry_perception" name="rasberry_perception_detection_backend" type="run_backend.sh" args="$(arg backend)" output="screen" required="true"/>
    </group>

    <arg name="image_ns" default="/camera/color"/>
    <arg name="depth_ns" default="/camera/aligned_depth_to_color"/>
    <arg name="score" default="0.5"/>

    <node pkg="rasberry_perception" name="rasberry_perception_detector" type="detection_client.py" output="screen" respawn="true">
        <param name="image_ns" type="str" value="$(arg image_ns)"/>
        <param name="depth_ns" type="str" value="$(arg depth_ns)"/>
        <param name="score" type="double" value="$(arg score)"/>
    </node>
</launch>
